### PR TITLE
Add robustness for fixed meshes in NearestNodeTransfer

### PR DESCRIPTION
**Pull request by lindsayad**
This addresses the trace-trace coupling failure in blue crab that
cropped up after the merging of #20451. I believe this failure
started occurring due to the bounding box changes in #20451. But
anyway in general there will be no guarantee that you can safely
index into the solution vector for any given dof when the vector
is distributed amongst processes. So we must guard against that

### CODE REVIEW

File: framework/src/transfers/MultiAppNearestNodeTransfer.C [hunk 1]
```diff
@@ -352,8 +352,8 @@ MultiAppNearestNodeTransfer::execute()
 
       if (_fixed_meshes)
       {
-        _cached_froms[pid].resize(qps.second.size());
-        _cached_dof_ids[pid].resize(qps.second.size());
+        _cached_froms[pid].resize(qps.second.size(), libMesh::invalid_uint);
+        _cached_dof_ids[pid].resize(qps.second.size(), DofObject::invalid_id);
```
Comment (loganharbour) [inline framework/src/transfers/MultiAppNearestNodeTransfer.C:356]: For thoroughness should we not instead do a fill after for all entries with invalid vals?

Comment (lindsayad) [reply framework/src/transfers/MultiAppNearestNodeTransfer.C:356]: I think in general there is no guarantee that the valid/invalid vector entries will be contiguous. There could be valid and invalid entries interlaced throughout the vector.

I'm not sure if that's what you were asking about/getting at though.

Comment (loganharbour) [reply framework/src/transfers/MultiAppNearestNodeTransfer.C:356]: What I mean is shouldn't we instead do:

```
auto & froms = _cached_froms[pid];
froms.resize(qps.second.size());
std::fill(froms.begin(), froms.end(), libMesh::invalid_uint);

auto & dof_ids = _cached_dof_ids[pid];
dof_ids.resize(qps.second.size());
std::fill(dof_ids.begin(), dof_ids.end(), DofObject::invalid_id);
```

Comment (lindsayad) [reply framework/src/transfers/MultiAppNearestNodeTransfer.C:356]: Yes that is a good idea ... that should be helpful if we ever do adaptivity in between these executes

**Review by loganharbour — APPROVED** [review]

**Pull request merged**